src/main/java/org/openstreetmap/atlas/checks/validation/areas/WaterbodyAndIslandSizeCheck.java

```diff
@@ -0,0 +1,336 @@
+        this.waterbodyMinimumArea = (double) configurationValue(configuration,
+                "area.waterbody.minimum.meters", WATERBODY_MIN_AREA_DEFAULT);
+        this.waterbodyMaximumArea = (double) configurationValue(configuration,
+                "area.waterbody.maximum.kilometers", WATERBODY_MAX_AREA_DEFAULT);
+        this.islandMinimumArea = (double) configurationValue(configuration,
+                "area.island.minimum.meters", ISLAND_MIN_AREA_DEFAULT);
+        this.islandMaximumArea = (double) configurationValue(configuration,
+                "area.island.maximum.kilometers", ISLAND_MAX_AREA_DEFAULT);
+        this.isletMinimumArea = (double) configurationValue(configuration,
+                "area.islet.minimum.meters", ISLET_MIN_AREA_DEFAULT);
+        this.isletMaximumArea = (double) configurationValue(configuration,
+                "area.islet.maximum.kilometers", ISLET_MAX_AREA_DEFAULT);
+    }
+
+    /**
+     * This function will validate if the supplied atlas object is valid for the check.
+     *
+     * @param object
+     *            the atlas object supplied by the Atlas-Checks framework for evaluation
+     * @return {@code true} if this object should be checked
+     */
+    @Override
+    public boolean validCheckForObject(final AtlasObject object)
+    {
+        // Object must be a MultiPolygon Relation with at least one member
+        return (object instanceof Relation && IS_MULTIPOLYGON_WATER_RELATION.test(object)
+                && !((Relation) object).members().isEmpty())
+                // An Area with place=islet, place=island, place=archipelago, or natural=water tag
+                || (object instanceof Area && (TagPredicates.IS_WATER_BODY.test(object)
+                        || IS_ISLET.test(object) || IS_ISLAND.test(object)))
+                        // And has yet to be examined
+                        && !this.isFlagged(object.getIdentifier());
+    }
+
+    /**
+     * This is the actual function that will check to see whether the object needs to be flagged.
+     *
+     * @param object
+     *            the atlas object supplied by the Atlas-Checks framework for evaluation
+     * @return an optional {@link CheckFlag} object that
+     */
+    @Override
+    protected Optional<CheckFlag> flag(final AtlasObject object)
+    {
+        if (object instanceof Area)
+        {
+            final Area area = (Area) object;
+            final double surfaceAreaMeters = area.asPolygon().surface().asMeterSquared();
+            final double surfaceAreaKilometers = area.asPolygon().surface().asKilometerSquared();
+            // Mark each object because we could potentially run into Multipolygon entities that get
+            // passed in as Areas
+            this.markAsFlagged(area.getIdentifier());
+
+            // natural=water tag
+            if (TagPredicates.IS_WATER_BODY.test(object))
+            {
+                if (surfaceAreaMeters < this.waterbodyMinimumArea)
```

Comment:
By using the `Distance` class and converting either the metres to kilometres or kilometres to metres, we would cut down on the amount of variables as well.

Reply:
Distance.kilometers(surfaceAreaMeters) ?